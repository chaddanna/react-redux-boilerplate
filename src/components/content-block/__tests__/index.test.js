import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Component from '..';

describe('Component', () => {
  let props;

  beforeEach(() => {
    props = {
      content: ''
    };
  });

  describe('Smoke tests', () => {
    it('snapshot test', () => {
      expect(
        renderer.create(<Component {...props} />).toJSON()
      ).toMatchSnapshot();
    });

    it('renders without crashing', () => {
      shallow(<Component {...props} />);
    });
  });

  describe('Mounted tests', () => {
    let mountedComponent;
    const getComponent = () => {
      if (!mountedComponent) {
        mountedComponent = mount(<Component {...props} />).find(Component);
      }

      return mountedComponent;
    };

    beforeEach(() => {
      mountedComponent = undefined;
    });

    it("always renders 'content-block'", () => {
      expect(getComponent().find('div.content-block')).toHaveLength(1);
    });

    describe("rendered 'content-block' with title set", () => {
      beforeEach(() => {
        props.content = 'Test';
      });

      it('sets content-block with Test', () => {
        expect(
          getComponent()
            .find('div.content-block')
            .text()
        ).toBe(`${props.content}`);
      });
    });
  });
});
