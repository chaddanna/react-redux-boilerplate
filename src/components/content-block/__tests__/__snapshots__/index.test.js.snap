// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Smoke tests snapshot test 1`] = `
<div
  className="content-block"
>
  <p>
    
  </p>
</div>
`;
